include/dlaf/matrix/tile.h

```diff
@@ -508,6 +510,15 @@ struct UnwrapFuture<hpx::future<Tile<T, D>>> {
   }
 };
 
+template <typename T>
+struct UnwrapFuture<hpx::future<dlaf::common::PromiseGuard<T>>> {
+  template <typename U>
+  static auto call(U&& u) {
+    auto t = u.get();
+    return t;
+  }
+};
+
```

Comment:
I'd suggest changing the specialization above this to just work for any future.

Also, I have no idea why I did `auto t = u.get(); return t;` instead of just `return u.get();`... I think that should be possible to change.

Finally, agreed on changing the name of `unwrapExtendTiles`. Perhaps `unwrapExtendArgs` or something similarly generic?

Reply:
An update on this: working on this PR I got misled and I thought that the PromiseGuard lifetime had to be extended. I was wrong (thanks @rasolca), because the PromiseGuard, which in this PR case provides a `Communicator` has to be alive during the posting of the MPI operation, and it is not needed for its completion.

Keeping it alive is not just superfluous, but it's wrong: by keeping it alive till the completion of the operation, it does not allow the pipeline to continue until the MPI operation is fully completed, leading to a sequential execution of async MPI calls (= no sense).

So, I just updated the existing `unwrapExtendTiles` with your suggestion about `return u.get()`.